### ISSUE TITLE
Data selection filter (map -> other dashboard components)

### ISSUE THREAD

**Issue by szilardhuber** [mapbox/mapboxgl-powerbi#2]
Users should be able to select a feature or set of features from the map using a bounding box selector.  Features in the selected region should be passed as a filter to other dashboard visualizaitons.  See this for an example: https://www.mapbox.com/mapbox-gl-js/example/using-box-queryrenderedfeatures/

**Comment by ryanbaumann** [comment]
This utility library may be helpful here: https://github.com/Microsoft/powerbi-visuals-utils-interactivityutils/blob/master/docs/api/interactivityService.md

**Comment by ryanbaumann** [comment]
Let's not enable this feature until we work on hover/highlight styles.  Moving to a lower priority and off milestone 2.

**Comment by ryanbaumann** [comment]
Fleshing out the details of this feature:

# To Do

- [ ] Finish the hover/highlight styles issue before starting this ticket - https://github.com/mapbox/mapboxgl-powerbi/issues/28
- [ ] Remove the `click` to zoom handler that exists on the map today.  
- [ ] Create a new js file to manage filter states, `filter.ts`
- [ ] In `filter.ts`, add a click-to-zoom handler on a map `click` event to filter handler (details below).
- [ ] Use the highlight features in a bounding box example to build a simple box selector triggered by holding shift + click + hold in `filter.js` - https://www.mapbox.com/mapbox-gl-js/example/using-box-queryrenderedfeatures/.  
- [ ] Use the Interactivity Service docs for Power BI Custom Visuals to syncronize dashboard filters across visualizations https://github.com/Microsoft/powerbi-visuals-utils-interactivityutils/blob/master/docs/api/interactivityService.md
- [ ] Persist the `hover` state for all features selected in a filter

**Comment by AndrewLockett** [comment]
Great to see the hover over 'tooltip' is now enabled for choropleth maps (1.2 beta). Will this filter feature work just when drawing a bounding box? Or will the user be able to select a polygon on the map and have that filter other viz on the dashboard (or ctrl click to select multiple areas)? The ArcGIS & ShapeMap viz work in this way. Any date estimate for these enhancements? Thanks

**Comment by ryanbaumann** [comment]
@AndrewLockett filters will work when drawing a bounding box and when clicking to select a polygon feature.

Dates are hard at this point - stay tuned on this ticket!  We're actively scoping the work on this feature now.

**Comment by AndrewLockett** [comment]
That is great news. I'll keep tuned for early beta releases. Happy to help with testing and provide feedback. Thanks for a great custom viz!

**Comment by samgehret** [comment]
This is mostly working.
Some bugs:
1. Shift + Highlight multiple states, it will only select one state.
2. When there are multiple drill levels, after selecting it drills down to the lowest level and will skip the second level.

**Comment by szilardhuber** [comment]
@samgehret There have been a couple of fixes regarding selection and the issues you described. Could you please check them again?

Regarding the specific issues:

1. We fixed that.
2. We could not reproduce. Could you please some give some more details if it is still present?
3. We were abel to reproduct it before our fixes and were not able to reproduce afterwards, but to be honest we don't know what could have been the root cause of this issue.
4. We don't quite understand this issue.

**Comment by szilardhuber** [comment]
@samgehret I think we addressed all issues now with 72bf16364eef65f48631fe19408419b1238830fa

**Comment by samgehret** [comment]
Hey team, great work. Things are working much smoother and the performance is improving every day. Most of the issues have been fixed. However a few remain.

1. When "drill mode" is on (the down arrow in the top right corner is highlighted), when I left-click on a state (or county, or any polygon), it should automatically drill into that selection and only that selection. When hovering over the "drilldown" selector, the tool tip reads "Drill mode is on: Click a data point to drill." From an end user perspective, I would expect a single click on a state will drill down to show the counties (or next level drill) for ONLY that selection. Additionally, if there are grids or other visualizations, they should be filtered on that click as well.
2. When "drill mode" is selected, and I shift + multiselect, it only drills into one state. This is an improvement over last time as now we are drilling into the correct level, however we are still only selecting a single state, when we should be selecting all of them. See the below gif. In this case, I would expect a drilldown into all the states in the selection, not just one state.
![image](https://user-images.githubusercontent.com/3859246/41172816-e1c3ef84-6b09-11e8-91d0-4d72b27e8620.png)

**Comment by samgehret** [comment]
@szilardhuber Thanks for the update. Regarding 1) Please make the change if you can. Regarding 2) We will open an issue with Power BI. Can you please create an issue in the open source repo with a code snippet, and we will flag it for Microsoft?

Also one more thing:
When drill mode is OFF, and a user single left clicks on a state (or any polygon). Can you implement the clicked polygon to remain selected and filter the grids. It works like this in ESRI. For example, if I left click on California it should look like this, filtering the grids on the right:
![image](https://user-images.githubusercontent.com/3859246/41185645-3c811c82-6b3f-11e8-8c8d-5e1aa3a359b2.png)
